buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesInfoTask.java

```diff
@@ -0,0 +1,315 @@
+
+        for (final Dependency dependency : runtimeDependencies) {
+            // we do not need compile-only dependencies here
+            if (compileOnlyArtifacts.contains(dependency.getGroup() + ":" +
+                                              dependency.getName() + ":" +
+                                              dependency.getVersion())) {
+                continue;
+            }
+            // only external dependencies are checked
+            if (dependency.getGroup() != null && dependency.getGroup().contains("org.elasticsearch")) {
+                continue;
+            }
+
+            final String url = createURL(dependency.getGroup(), dependency.getName(), dependency.getVersion());
+            final String dependencyName = DependencyLicensesTask.getDependencyName(mappings, dependency.getName());
+            logger.info("mapped dependency " + dependency.getGroup() + ":" + dependency.getName() + " to " + dependencyName + " for license info");
+
+            final String licenseType = getLicenseType(dependency.group, dependencyName);
+            output.append(dependency.getGroup() + ":" + dependency.getName() + "," + dependency.getVersion() + "," + url + "," + licenseType + "\n");
+
+        }
+        Files.write(outputfile.toPath(), output.toString().getBytes(StandardCharset.UTF_8));
+    }
+    
+    /**
+     * Create an URL on <a href="https://repo1.maven.org/maven2/">Maven Central</a>
+     * based on dependency coordinates.
+     */
+    protected String createURL(final String group, final String name, final String version){
+        final String baseURL = "https://repo1.maven.org/maven2";
+        return baseURL + "/" + group.replaceAll("\\.", "/") + "/" + name + "/" + version;
+    }
+    
+    /**
+     * Read the LICENSE file associated with the dependency and determine a license type.
+     *
+     * The license type is one of the following values:
+     * <u>
+     *     <li><em>UNKNOWN</em> if LICENSE file is not present for this dependency.</li>
+     *     <li><em>one SPDX identifier</em> if the LICENSE content matches with an SPDX license.</li>
+     *     <li><em>Custom;URL</em> if it's not an SPDX license,
+     *          URL is the Github URL to the LICENSE file in elasticsearch repository.</li>
+     * </ul>
+     *
+     * @param group dependency group
+     * @param name dependency name
+     * @return SPDX identifier, UNKNOWN or a Custom license
+     */
+    protected String getLicenseType(final String group, final String name) throws IOException {
+        File license;
+
+        if (licensesDir.exists()) {
+            File[] matchedfiles = licensesDir.listFiles((dir, name) -> name.contains("-LICENSE"));
+            for(File file : matchedfiles) {
+                String prefix = file.getName().split("-LICENSE.*")[0];
+                if (group.contains(prefix) || name.contains(prefix)) {
+                    license = file.getAbsoluteFile();
```

Comment:
I know this is same as existing code but, makes me curious. 

I wonder how this `loop` works? When the `if (group.contains(prefix) || name.contains(prefix))` is true more than `once` then previous value of `license` is replaced with `new one` 🤔If it is `true` only `once` then we should `return` early after assigning `license` value. No? 😕

Reply:
I'm not sure, but I wonder if there could be more than one license file for a dependency? How would we know without looping through all of them? And if there are, looping through all of them and sticking to the last one makes as much sense as any other.